### ISSUE TITLE
Key discovery proposal: feedback requested

### ISSUE THREAD

**Issue by csharrison** [WICG/attribution-reporting-api#864]
We recently published a proposal for "key discovery" for summary reports, which allows queries which do not pre-declare buckets:
https://github.com/WICG/attribution-reporting-api/blob/main/aggregate_key_discovery.md

We're opening this issue to solicit general feedback on the proposal.
cc @hidayetaksu

**Comment by alfrednfwong** [comment]
Can you always return an "others" bucket that sums the noisy totals of all the buckets dropped because they didnt make the threshold, and those that are undeclared and are not even in the key-mask?

That way, if we sum all the reported bucket totals, including the "others", we'll have an unbiased estimate of the true total. Also, the "others" bucket allows us to know how much we are missing with the declared buckets, and can adapt our aggregation key strategy accordingly.

**Comment by keke123** [comment]
We do not have a proposal in place for returning an "others" bucket yet but it is something we're discussing internally. It is technically feasible to return this but it would need to be designed to be privacy preserving. One possible solution once we have requerying would be to allow adtechs to split their allocated privacy budget between the key discovery query (ie the results above the threshold) and a new query type for metadata without results (ex: count of total buckets, sum of values) to get the unbiased totals. This way adtechs could get both sets of information in a privacy preserving way. 

Note that if the adtech already knows the buckets, they can send a separate query for the buckets that were dropped to get the unbiased totals. A few follow up questions:

- Is having the information on dropped buckets a blocker to key discovery adoption? 
- How would you prioritize the information about dropped buckets (ex: count of buckets dropped vs noisy totals of buckets dropped) you would want to receive? 

#583

**Comment by alfrednfwong** [comment]
> Is having the information on dropped buckets a blocker to key discovery adoption?
No. Key discovery or not, we don't have an unbiased estimate of the total without the "others" bucket.
But not having the information on dropped buckets dissuades us from relying on the summary reports.

>  How would you prioritize the information about dropped buckets (ex: count of buckets dropped vs noisy totals of buckets dropped) you would want to receive?
Total is more important than count of buckets dropped

**Comment by CGossec** [comment]
We (Criteo) would tentatively welcome this change.

It starts to provide a potential solution to the original problem we raised in [Knowing the source site in the aggregation API / aggregate queries need key discovery mechanism · Issue #583](https://github.com/WICG/attribution-reporting-api/issues/583), which is still an issue right now.

That being said, we also believe that the need for source-site information is a relatively universal need for adtechs in general. We would be very interested in hearing more about what you have in mind when you mention in the Future Improvements of the proposal:

> Along the lines of [issue 583](https://github.com/WICG/attribution-reporting-api/issues/583), consider allowing the browser to embed known values in the encrypted payload (e.g. the source site) to avoid needing a custom encoding.

Finally, you mention a shift to the Truncated Laplace distribution, and the impact this new distribution is to have on the noise is a bit unclear. Can you give details on the distribution itself, notably on the final target values of the different parameters (epsilon, delta, any others) you are thinking about when mentioning it? We strongly believe having more visibility on the changes this proposal brings noise-wise is important.

**Comment by KiHunCho** [comment]
Hello,
I’m reviewing the explainers to evaluate the adoption of the Attribution Reporting API.
I have a question regarding the proposed “key-discovery” feature mentioned in the [explainer](https://github.com/WICG/attribution-reporting-api/blob/main/aggregate_key_discovery.md).

In the key-discovery explainer, the following is stated:
“_For any other (non pre-declared) bucket matching the key-mask provided by the ad tech_” 

If I set the _pre-declared buckets set_ to empty and the _key-mask_ to "0x0...01" (only the rightmost bit is set to 1), does that mean the number of buckets matched by the key-mask would be 2^127?

In #256, I understood that “Group By” operations on the aggregatable reports uploaded to the Aggregation Service could be enabled through the proposed key-discovery.

However, based on what I see in the key-discovery explainer, it appears that “Group By” functionality is not supported — so I wanted to confirm if my understanding is correct.

Thank you!
